src/debugger/RunDebugAdapter.ts

```diff
@@ -0,0 +1,121 @@
+import * as vscode from "vscode";
+import { DebugProtocol } from "@vscode/debugprotocol";
+
+export class Message implements DebugProtocol.ProtocolMessage {
+  seq: number;
+  type: string;
+
+  public constructor(type: string) {
+    this.seq = 0;
+    this.type = type;
+  }
+}
+
+export class Event extends Message implements DebugProtocol.Event {
+  event: string;
+
+  public constructor(event: string, body?: any) {
+    super("event");
+    this.event = event;
+    if (body) {
+      (<any>this).body = body;
+    }
+  }
+}
+
+class ExitedEvent extends Event implements DebugProtocol.ExitedEvent {
+  body: {
+    exitCode: number;
+  };
+
+  public constructor(exitCode: number) {
+    super("exited");
+    this.body = {
+      exitCode: exitCode,
+    };
+  }
+}
+
+export class TerminatedEvent
+  extends Event
+  implements DebugProtocol.TerminatedEvent
+{
+  public constructor(restart?: any) {
+    super("terminated");
+    if (typeof restart === "boolean" || restart) {
+      const e: DebugProtocol.TerminatedEvent = this;
+      e.body = {
+        restart: restart,
+      };
+    }
+  }
+}
+
+class Response extends Message implements DebugProtocol.Response {
+  request_seq: number;
+  success: boolean;
+  command: string;
+
+  public constructor(request: DebugProtocol.Request, message?: string) {
+    super("response");
+    this.request_seq = request.seq;
+    this.command = request.command;
+    if (message) {
+      this.success = false;
+      (<any>this).message = message;
+    } else {
+      this.success = true;
+    }
+  }
+}
+
+export class RunDebugAdapter implements vscode.DebugAdapter {
```

Comment:
So this is a fake server so that we can run in terminal from config. Unfortunately, we need to do it via DAP, so the easiest way seemed just faking it. One minor issue is that the icons for debugging show up for a second.